lib/manager/common.ts

```diff
@@ -0,0 +1,216 @@
+import { ReleaseType } from 'semver';
+import { RangeStrategy } from '../versioning';
+
+export type Result<T> = T | Promise<T>;
+
+export interface ManagerConfig {
+  binarySource?: string;
+  localDir?: string;
+  registryUrls?: (string | Registry)[];
+}
+
+export interface ManagerData<T> {
+  managerData?: T;
+}
+
+export interface ExtractConfig extends ManagerConfig {
+  endpoint?: string;
+  global?: any;
+  gradle?: { timeout?: number };
+  ignoreNpmrcFile?: boolean;
+
+  skipInstalls?: boolean;
+  versionScheme?: string;
+}
+
+export interface UpdateArtifactsConfig extends ManagerConfig {
+  isLockFileMaintenance?: boolean;
+  compatibility?: Record<string, string>;
+  cacheDir?: string;
+  postUpdateOptions?: string[];
+}
+
+export interface PackageUpdateConfig {
+  currentValue?: string;
+  rangeStrategy?: RangeStrategy;
+  supportPolicy?: string[];
+}
+
+export interface PackageUpdateResult {
+  newValue: string[];
+  newMajor: string;
+  isRange: boolean;
+  sourceUrl: string;
+}
+
+export interface RangeConfig<T = Record<string, any>> extends ManagerData<T> {
+  composerJsonType?: 'composer-plugin' | 'library' | 'metapackage' | 'project';
+  currentValue?: string;
+  depName?: string;
+  depType?: string;
+  manager?: string;
+  packageJsonType?: 'app' | 'library';
+  rangeStrategy: RangeStrategy;
+}
+
+export interface Registry {
+  type?: string;
+  url: string;
+}
+
+export interface NpmLockFiles {
+  yarnIntegrity?: boolean;
+  yarnLock?: string;
+  packageLock?: string;
+  shrinkwrapJson?: string;
+  pnpmShrinkwrap?: string;
+  npmLock?: string;
+  lernaDir?: string;
+}
+
+export interface PackageFile<T = Record<string, any>>
+  extends NpmLockFiles,
+    ManagerData<T> {
+  hasYarnWorkspaces?: boolean;
+  internalPackages?: string[];
+  compatibility?: Record<string, string>;
+  datasource?: string;
+  registryUrls?: (string | Registry)[];
+  deps: PackageDependency[];
+  ignoreNpmrcFile?: boolean;
+  lernaClient?: string;
+  lernaPackages?: string[];
+  manager?: string;
+  mavenProps?: Record<string, any>;
```

Comment:
Is there any automated way to determine if fields like this are. no longer needed?

Reply:
no, but vscode will show the reference count, if you use [codelense](https://code.visualstudio.com/blogs/2017/02/12/code-lens-roundup)

![image](https://user-images.githubusercontent.com/1798109/61827043-f9f3a380-ae63-11e9-96bc-5132449064f3.png)

you can click on it to show where it is used.